wp/theme/AbstractArchive.php

```diff
@@ -79,6 +78,28 @@ function sort_posts($a, $b){
 		return array_search($b->ID, $this->ids) > array_search($a->ID, $this->ids) ? -1 : 1;
 	}
 
+
+	function filter_frontend_main_query($allowed,$wp_query)	{
+		return $allowed && $wp_query->is_main_query() && !is_admin();
+	}
+
+	function filter_singular_query($allowed,$wp_query) {
+		return $allowed && !$this->attempted;
+	}
+
+	/**
+	 * Removes any namespaces and transforms the class name into lowercase for use in filters
+	 * 
+	 * @return string
+	 */
+	function get_filter_name() {
```

Comment:
not a fan of having filters depend on class names, as they can change at any time. better to maybe put common logic in abstract and have child classes call it instead with its own name